code/game/objects/structures/dresser.dm

```diff
@@ -19,6 +19,36 @@
 	new /obj/item/stack/sheet/mineral/wood(drop_location(), 10)
 	qdel(src)
 
+/obj/structure/dresser/Initialize(mapload)
+	. = ..()
+	var/datum/component/storage/STR = AddComponent(/datum/component/storage/concrete)
+	STR.max_w_class = WEIGHT_CLASS_SMALL
+	STR.max_combined_w_class = 80
+	STR.max_items = 40
+
+	if(mapload)
+		generate_underwear()
+
+#define UNDIE_AMOUNT 11
+/obj/structure/dresser/proc/generate_underwear()
+	var/static/list/possible_underwear = subtypesof(/obj/item/clothing/underwear/briefs)
+	var/static/list/possible_undershirt = subtypesof(/obj/item/clothing/underwear/shirt)
+	var/static/list/possible_socks = subtypesof(/obj/item/clothing/underwear/socks)
```

Comment:
`typesof()` root path is still valid

Reply:
Wasn't sure so I used subtypes

Reply:
Happy to help

Reply:
also, i'd recommend storing it in a global list, or at least a static list outside of the proc